### PR TITLE
Expand ~ as user's home directory

### PR DESCRIPTION
**Pull request by nunomdc**
Should fix issue https://github.com/alichtman/shallow-backup/issues/155.

### CODE REVIEW

File: shallow_backup/cli.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def cli(add, rm, show, all, dotfiles, configs, packages, fonts, old_path, new_pa
 			print_red_bold("Removed config file...")
 		elif destroy_backup:
 			backup_home_path = get_config()["backup_path"]
+			backup_home_path = os.path.abspath(os.path.expanduser(backup_home_path))
```
Comment (alichtman) [inline shallow_backup/cli.py]: Let's extract `os.path.abspath(os.path.expanduser(VAR))` to a function that lives in `utils.py` to reduce code repetition.

**Comment by nunomdc** [comment]
I can see where you're coming from, although if we're going to move this into a separate function then we could also explore the possibility to expand system environment variables on path inputs

**Comment by alichtman** [comment]
> we could also explore the possibility to expand system environment variables on path inputs

I'm confused what you mean by this. Can you provide an example?

File: shallow_backup/utils.py [hunk 2]
```diff
@@ -110,3 +110,12 @@ def home_prefix(path):
 	"""
 	home_path = os.path.expanduser('~')
 	return os.path.join(home_path, path)
+
+def expand_path_to_absolute(path):
```
Comment (alichtman) [inline shallow_backup/utils.py]:
```suggestion
def expand_to_abs_path(path):
```

Comment (alichtman) [reply shallow_backup/utils.py]: Just to keep the names as short as possible, but still descriptive.

**Comment by nunomdc** [comment]
> > we could also explore the possibility to expand system environment variables on path inputs
> 
> I'm confused what you mean by this. Can you provide an example where this would be useful?

For instance, accepting the following inputs:

- $HOME/shallow-backup
- $MY_BACKUP_DIR/shallow-backup

**Comment by alichtman** [comment]
Yes, that's an awesome idea. Do you want to implement that?

**Pull request merged**